test/common/network/address_impl_test.cc

```diff
@@ -392,6 +392,62 @@ TEST(AddressFromSockAddr, Pipe) {
 #endif
 }
 
+// Test comparisons between all the different (known) test classes.
+TEST(MixedAddresses, Equality) {
```

Comment:
nit: Is it possible to somehow use TEST_P to do all of the mix and match logic? I'm guessing it is, but I'm not a gtest expert. Not a big deal if it's not easy.

Reply:
Glancing at the docs for TEST_P (https://github.com/google/googletest/blob/master/googletest/docs/advanced.md, search for TEST_P) I don't see how to do this.  The problem is that we want a cross-product of all the possible addresses, and TEST_P only takes a single list of parameters.  I could make a structure that contained both things to compare, but I don't see a way to auto-generate that structure, which means I'm effectively replacing two for loops with a hardcoded list.  That doesn't strike me as a step forward for readability or maintainability.

Do you see anything I'm missing?

Reply:
I thin you can do something with `testing::Combine` like here: https://github.com/envoyproxy/envoy/blob/master/test/common/http/http2/codec_impl_test.cc#L669 but I'm not completely sure. It's not a big deal either way.

Reply:
Ah, neat.  Done.  (I'm not *sure* it was worth it, but I learned something so I'll call that a win :-}.)